serde_v8/src/magic/zero_copy_buf.rs

```diff
@@ -19,118 +19,87 @@ use super::transl8::FromV8;
 /// `let copy = Vec::from(&*zero_copy_buf);`
 #[derive(Clone)]
 pub struct ZeroCopyBuf {
-  backing_store: v8::SharedRef<v8::BackingStore>,
-  byte_offset: usize,
-  byte_length: usize,
+  pub(crate) store: v8::SharedRef<v8::BackingStore>,
+  pub(crate) range: Range<usize>,
 }
 
 unsafe impl Send for ZeroCopyBuf {}
 
 impl ZeroCopyBuf {
   pub fn from_buffer(
     buffer: v8::Local<v8::ArrayBuffer>,
-    byte_offset: usize,
-    byte_length: usize,
+    range: Range<usize>,
   ) -> Result<Self, v8::DataError> {
-    let backing_store = buffer.get_backing_store();
-    match backing_store.is_shared() {
-      true => Err(v8::DataError::BadType {
+    let store = buffer.get_backing_store();
+    if store.is_shared() {
+      return Err(v8::DataError::BadType {
         actual: "shared ArrayBufferView",
         expected: "non-shared ArrayBufferView",
-      }),
-      false => Ok(Self {
-        backing_store,
-        byte_offset,
-        byte_length,
-      }),
+      });
     }
+    Ok(Self { store, range })
   }
 
-  pub fn from_view(
-    scope: &mut v8::HandleScope,
-    view: v8::Local<v8::ArrayBufferView>,
-  ) -> Result<Self, v8::DataError> {
+  fn as_slice(&self) -> &[u8] {
+    unsafe { &*(&self.store[self.range.clone()] as *const _ as *const [u8]) }
+  }
+
+  #[allow(clippy::cast_ref_to_mut)]
+  fn as_slice_mut(&mut self) -> &mut [u8] {
+    unsafe { &mut *(&self.store[self.range.clone()] as *const _ as *mut [u8]) }
+  }
+}
+
+pub(crate) fn to_ranged_buffer<'a, 's>(
+  scope: &mut v8::HandleScope<'s>,
+  value: v8::Local<'a, v8::Value>,
+) -> Result<(v8::Local<'a, v8::ArrayBuffer>, Range<usize>), v8::DataError> {
+  if value.is_array_buffer_view() {
+    let view: v8::Local<v8::ArrayBufferView> = value.try_into()?;
+    let (offset, len) = (view.byte_offset(), view.byte_length());
     let buffer = view.buffer(scope).ok_or(v8::DataError::NoData {
       expected: "view to have a buffer",
     })?;
-    Self::from_buffer(buffer, view.byte_offset(), view.byte_length())
+    let buffer = unsafe { std::mem::transmute(buffer) }; // lifetime hack
+    return Ok((buffer, offset..offset + len));
   }
+  let b: v8::Local<v8::ArrayBuffer> = value.try_into()?;
+  let b = unsafe { std::mem::transmute(b) }; // lifetime hack
```

Comment:
`// SAFETY`

Reply:
Removed unsafe in [50b41b5](https://github.com/denoland/deno/pull/14102/commits/50b41b5ba5f637f0dc21233c79c044da2477f8c9)